sql/core/src/main/scala/org/apache/spark/sql/execution/Generate.scala

```diff
@@ -34,17 +36,22 @@ import org.apache.spark.sql.catalyst.expressions._
 @DeveloperApi
 case class Generate(
     generator: Generator,
+    tblAlias: Option[String],
     join: Boolean,
     outer: Boolean,
     child: SparkPlan)
   extends UnaryNode {
 
   // This must be a val since the generator output expr ids are not preserved by serialization.
```

Comment:
I'm tempted to remove all this complicated logic and just make the attributes produced an argument to the `Generate` operator.